plugin/metrics/setup.go

```diff
@@ -50,6 +51,27 @@ func setup(c *caddy.Controller) error {
 		return nil
 	})
 
+	c.OnRestart(func() error {
+		vars.PluginEnabled.Reset()
+		return nil
+	})
+
+	c.OnStartup(func() error {
+		conf := dnsserver.GetConfig(c)
+		plugins := conf.Handlers()
```

Comment:
Slightly shorter: `conf := dnsserver.GetConfig(c).Handlers()`

Reply:
If I don't need a for loop to iterate the hosts to `resolveTCPAddr` later, then I will not need this.